Research/ResearchUI/iOS/Views/RSDVerticalGridCollectionView.swift

```diff
@@ -0,0 +1,191 @@
+//
+// 2.  Redistributions in binary form must reproduce the above copyright notice,
+// this list of conditions and the following disclaimer in the documentation and/or
+// other materials provided with the distribution.
+//
+// 3.  Neither the name of the copyright holder(s) nor the names of any contributors
+// may be used to endorse or promote products derived from this software without
+// specific prior written permission. No license is granted to the trademarks of
+// the copyright holders even if such marks are included in this software.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+
+import Foundation
+
+/// The 'RSDVerticalGridCollectionView' is a normal collection view that forces a
+/// 'RSDVerticalGridCollectionViewFlowLayout' as the colelction view layout
+open class RSDVerticalGridCollectionView: UICollectionView {
+    
+    /// Force cast of collectin view layout as a RSDVerticalGridCollectionViewFlowLayout
+    open var gridLayout: RSDVerticalGridCollectionViewFlowLayout {
+        return self.collectionViewLayout as! RSDVerticalGridCollectionViewFlowLayout
+    }
+    
+    public override init(frame: CGRect, collectionViewLayout layout: UICollectionViewLayout) {
+        
+        super.init(frame: frame, collectionViewLayout: layout)
+                        
+        if !layout.isKind(of: RSDVerticalGridCollectionViewFlowLayout.self) {
+            debugPrint("WARNING: UICollectionViewLayout must be RSDVerticalGridCollectionViewFlowLayout, provided one will be overwritten")
+            self.initDefaultFlowLayout()
+        }
+    }
+
+    public required init?(coder: NSCoder) {
+        super.init(coder: coder)
+        self.initDefaultFlowLayout()
+    }
+    
+    fileprivate func initDefaultFlowLayout() {
+        let layout = RSDVerticalGridCollectionViewFlowLayout()
+        self.collectionViewLayout = layout
+    }
+}
+
+/// The 'RSDVerticalGridCollectionViewFlowLayout' creates a centered grid layout in conjunction with
+/// the 'RSDVerticalGridCollectionView'.  As the class name implies, this class only works currently in a vertical orientation.
+open class RSDVerticalGridCollectionViewFlowLayout: UICollectionViewFlowLayout {
```

Comment:
You need a lot more documentation on how to use this class. The fact that you are co-opting the definition of "section" to mean "grid row" is really confusing and not self-evident. You should be including an explanation on how to use this layout.

Reply:
> Added @stockards as a reviewer.
> 
> I will add additional comments once I look at the code. That said, from the images, things that I noticed:
> 
> ### Definitely a bug
> 1. iPhone11ProMax_6items - "Comfortable" does not fit on one line and the "e" is wrapped to the next line. Is the text font the correct size? I thought I fixed that?
> 
> ### Maybe a bug?
> 1. iPhone11ProMax_2items - The text should break as "Comfortable\nPlace to Sit" rather than "Comfortable Place\nTo Sit" (Or did that change?)
> 2. iPhone11ProMax_4items - Should probably be 2/2 rather than 3/1
> 3. iPhone11ProMax - When there are two rows, the vertical spacing looks a bit tight

Fixed the main bug

<img width="266" alt="Screen Shot 2019-12-09 at 4 47 44 PM" src="https://user-images.githubusercontent.com/5590748/70475963-b19bf180-1aa3-11ea-8fe0-46dd959fb80b.png">